rocksdb_replicator/tests/rocksdb_replicator_test.cpp

```diff
@@ -485,6 +489,101 @@ TEST(RocksDBReplicatorTest, 1_leader_2_followers_deadlock) {
   EXPECT_EQ(0, db_slave_2->GetLatestSequenceNumber());
 }
 
+TEST(RocksDBReplicatorTest, 1_master_1_slave_replication_mode_2) {
+  // enable 2-ACK mode
+  FLAGS_replicator_replication_mode = 2;
+
+  FLAGS_replicator_timeout_ms = 100;
+  FLAGS_replicator_timeout_degraded_ms = 5;
+
+  // setup a shard with 1 master and 1 slave
+  int16_t master_port = 9092;
+  int16_t slave_port = 9093;
+  Host master(master_port);
+  Host slave(slave_port);
+
+  auto db_master = cleanAndOpenDB("/tmp/db_master");
+  auto db_slave = cleanAndOpenDB("/tmp/db_slave");
+
+  RocksDBReplicator::ReplicatedDB* replicated_db_master;
+  RocksDBReplicator::ReplicatedDB* replicated_db_slave;
+
+  EXPECT_EQ(master.replicator_->addDB("shard1", db_master, ReplicaRole::LEADER, folly::SocketAddress(),
```

Comment:
Would it be possible to add checks to make sure another shard `shard2` remains unaffected?

Reply:
Good idea, added tests on that.